### PR TITLE
Added save argument to Beacon state module

### PR DESCRIPTION
**Pull request by rdutch**
### What does this PR do?
Adds a save is True/False argument to the beacon.present and the beacon.absent state functions.
This argument will save the current beacons in memory to the beacons.conf file.

Removed the merge section from the beacons/__init__.py. Used in the list beacons functions, now only lists beacons loaded in memory.
### What issues does this PR fix or reference?
ZD 1758

### Tests written?

Was tested on 2017.7.1 with CentOs 7.3

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Review by garethgreenaway — APPROVED** [review]

File: salt/beacons/__init__.py [hunk 1]
```diff
@@ -205,10 +205,10 @@ def list_beacons(self):
         '''
         # Fire the complete event back along with the list of beacons
         evt = salt.utils.event.get_event('minion', opts=self.opts)
-        b_conf = self.functions['config.merge']('beacons')
-        if not isinstance(self.opts['beacons'], dict):
-            self.opts['beacons'] = {}
-        self.opts['beacons'].update(b_conf)
+        #b_conf = self.functions['config.merge']('beacons')
```
Comment (cachedout) [inline salt/beacons/__init__.py]: Please remove these comments. We don't like to have dead code in the codebase. Thanks.

**Review by cachedout — CHANGES_REQUESTED** [review]

File: salt/states/beacon.py [hunk 2]
```diff
@@ -128,6 +141,10 @@ def absent(name, **kwargs):
     else:
         ret['comment'].append('{0} not configured in beacons'.format(name))
 
+    if save == True:
```
Comment (cachedout) [inline salt/states/beacon.py]: This should be `if save:`

File: salt/states/beacon.py [hunk 3]
```diff
@@ -128,6 +141,10 @@ def absent(name, **kwargs):
...
+        result = __salt__['beacons.save']()
+        ret['comment'].append('Beacons saved'.format(name))
```
Comment (cachedout) [inline salt/states/beacon.py]: This should be `ret['comment'].append('Beacon {0} saved'.format(name))`

File: salt/states/beacon.py [hunk 4]
```diff
@@ -90,16 +95,24 @@ def present(name,
             else:
                 ret['comment'].append('Adding {0} to beacons'.format(name))
 
+    if save == True:
+        result = __salt__['beacons.save']()
+        ret['comment'].append('Beacons saved'.format(name))
```
Comment (cachedout) [inline salt/states/beacon.py]: This should be `ret['comment'].append('Beacon {0} saved'.format(name))`

File: salt/states/beacon.py [hunk 5]
```diff
@@ -61,7 +65,7 @@ def present(name,
         else:
             if 'test' in __opts__ and __opts__['test']:
                 kwargs['test'] = True
-                result = __salt__['beacons.modify'](name, beacon_data)
+                result = __salt__['beacons.modify'](name, beacon_data, **kwargs)
```
Comment (cachedout) [inline salt/states/beacon.py]: I don't think this is what we want here. We already have kwargs in the `beacon_data` list up on L55. Instead, we just need to merge the `save` option into that structure.

**Review by cachedout — CHANGES_REQUESTED** [review]

File: salt/states/beacon.py (hunk 5 above)
Comment (rdutch) [reply salt/states/beacon.py]: This was not a change by me, but took it out now. You are right, should not be there since is already in beacon_data.

**Comment by cachedout** [comment]
@rdutch This still won't pass automated checks until the change to `if save:` is made.

**Review by cachedout — APPROVED** [review]

**Pull request merged**